wazo_agentd/service/action/add.py

```diff
@@ -22,24 +23,33 @@ def add_agent_to_queue(self, agent_status, queue):
     def _update_asterisk(self, agent_status, queue):
         member_name = format_agent_member_name(agent_status.agent_number)
         skills = format_agent_skills(agent_status.agent_id)
-        response = self._amid_client.action(
-            'QueueAdd',
-            {
-                'Queue': queue.name,
-                'Interface': agent_status.interface,
-                'MemberName': member_name,
-                'StateInterface': agent_status.state_interface,
-                'Penalty': queue.penalty,
-                'Skills': skills,
-            },
-        )
-        if response[0]['Response'] != 'Success':
+        try:
+            self._amid_client.action(
+                'QueueAdd',
+                {
+                    'Queue': queue.name,
+                    'Interface': agent_status.interface,
+                    'MemberName': member_name,
+                    'StateInterface': agent_status.state_interface,
+                    'Penalty': queue.penalty,
+                    'Skills': skills,
+                    **self._format_pause_status(agent_status),
+                },
+            )
+        except AmidProtocolError as e:
             logger.warning(
-                'Failure to add interface %r to queue %r',
+                'Failure to add interface %s to queue %s: %s',
                 agent_status.interface,
                 queue.name,
+                e,
             )
 
     def _update_agent_status(self, agent_status, queue):
         with db_utils.session_scope():
             self._agent_status_dao.add_agent_to_queues(agent_status.agent_id, [queue])
+
+    def _format_pause_status(self, agent_status):
+        value = {'Paused': '1' if agent_status.paused else '0'}
+        if (reason := agent_status.paused_reason) and agent_status.paused:
+            value['Reason'] = reason
+        return value
```

Comment:
Mainly depending on how `paused_reason` is handled down the line, I'd suggest something like that, in KISS spirit:
```suggestion
    def _format_pause_status(self, agent_status):
        if agent_status.paused:
            return {
                'Paused': '1',
                'Reason': agent_status.paused_reason if agent_status.paused_reason else '',
            }
        else:
            return {'Paused': '0'}
```

Reply:
Keeping it as is, just because of the unknown behaviours down the line if `Reason` is present but empty or `''`